client/my-sites/email/inbox/mailbox-selection-list/index.jsx

```diff
@@ -0,0 +1,177 @@
+import { Button, Card, Gridicon } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import PropTypes from 'prop-types';
+import React, { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import googleWorkspaceIcon from 'calypso/assets/images/email-providers/google-workspace/icon.svg';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { useGetMailboxes } from 'calypso/data/emails/use-get-mailboxes';
+import {
+	getEmailAddress,
+	isEmailForwardAccount,
+	isGoogleEmailAccount,
+	isTitanMailAccount,
+} from 'calypso/lib/emails';
+import { getGmailUrl } from 'calypso/lib/gsuite';
+import { getTitanEmailUrl } from 'calypso/lib/titan';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import ProgressLine from './progress-line';
+
+import './style.scss';
+
+const getExternalUrl = ( mailbox ) => {
+	if ( isTitanMailAccount( mailbox ) ) {
+		return getTitanEmailUrl( getEmailAddress( mailbox ) );
+	}
+
+	if ( isGoogleEmailAccount( mailbox ) ) {
+		return getGmailUrl( getEmailAddress( mailbox ) );
+	}
+
+	return '';
+};
+
+const MailboxItemIcon = ( { mailbox } ) => {
+	const translate = useTranslate();
+
+	if ( isTitanMailAccount( mailbox ) ) {
+		return (
+			<span className="mailbox-selection-list__icon-circle">{ mailbox.mailbox[ 0 ] ?? 'T' }</span>
+		);
+	}
+
+	if ( isGoogleEmailAccount( mailbox ) ) {
+		return <img src={ googleWorkspaceIcon } alt={ translate( 'Google Workspace icon' ) } />;
+	}
+
+	return null;
+};
+
+MailboxItemIcon.propType = {
+	mailbox: PropTypes.object.isRequired,
+};
+
+const MailboxItem = ( { mailbox } ) => {
+	if ( isEmailForwardAccount( mailbox ) ) {
+		return null;
+	}
+
+	return (
+		<Card
+			className="mailbox-selection-list__mailbox"
+			href={ getExternalUrl( mailbox ) }
+			target="external"
+		>
+			<span className="mailbox-selection-list__icon">
+				<MailboxItemIcon mailbox={ mailbox } />
+			</span>
+			<div className="mailbox-selection-list__domain">
```

Comment:
I think this should be `mailbox-selection-list__mailbox` or maybe even `mailbox-selection-list__email`. The value isn't a domain.
```suggestion
			<div className="mailbox-selection-list__mailbox">
```